components/admin_console/brand_image_setting/brand_image_setting.jsx

```diff
@@ -62,86 +70,129 @@ export default class BrandImageSetting extends React.PureComponent {
 
             reader.readAsDataURL(this.state.brandImage);
         }
+
+        if (this.props.saving && this.state.brandImage && !this.props.disabled) {
+            this.handleImageSubmit();
+        }
+
+        if (this.props.saving && this.state.deleteBrandImage && !this.props.disabled) {
+            this.handleImageDelete();
+        }
     }
 
     handleImageChange() {
         const element = $(this.refs.fileInput);
-
         if (element.prop('files').length > 0) {
+            this.props.onChange('saveNeeded', 'config');
```

Comment:
Maybe we could pass in a separate `setSaveNeeded` prop to use for this instead of using `onChange`? This works, but it isn't really how `onChange` was intended to work, so it could get broken accidentally in the future.

Alternatively, if we end up going with something like `registerSaveAction` like I mentioned above, that could set `saveNeeded` for you when you call it.

Reply:
Resolved!